integration-tests/src/main/java/io/druid/testing/utils/TestQueryHelper.java

```diff
@@ -0,0 +1,149 @@
+    List<QueryWithResults> queries =
+        jsonMapper.readValue(
+            TestQueryHelper.class.getResourceAsStream(filePath),
+            new TypeReference<List<QueryWithResults>>()
+            {
+            }
+        );
+    testFailureQueries(queries, timesToRun);
+  }
+
+  public void testQueriesFromFile(String filePath, int timesToRun) throws Exception
+  {
+    testQueriesFromFile(getBrokerURL(), filePath, timesToRun);
+  }
+
+  public void testQueriesFromFile(String url, String filePath, int timesToRun) throws Exception
+  {
+    LOG.info("Starting query tests for [%s]", filePath);
+    List<QueryWithResults> queries =
+        jsonMapper.readValue(
+            TestQueryHelper.class.getResourceAsStream(filePath),
+            new TypeReference<List<QueryWithResults>>()
+            {
+            }
+        );
+    testQueries(url, queries, timesToRun);
+  }
+
+  public void testQueriesFromString(String str, int timesToRun) throws Exception
+  {
+    testQueriesFromString(getBrokerURL(), str, timesToRun);
+  }
+
+  public void testQueriesFromString(String url, String str, int timesToRun) throws Exception
+  {
+    LOG.info("Starting query tests using\n%s", str);
+    List<QueryWithResults> queries =
+        jsonMapper.readValue(
+            str,
+            new TypeReference<List<QueryWithResults>>()
+            {
+            }
+        );
+    testQueries(url, queries, timesToRun);
+  }
+
+  private void testQueries(String url, List<QueryWithResults> queries, int timesToRun) throws Exception
+  {
+    for (int i = 0; i < timesToRun; i++) {
+      LOG.info("Starting Iteration %d", i);
+
+      boolean failed = false;
+      for (QueryWithResults queryWithResult : queries) {
+        LOG.info("Running Query %s", queryWithResult.getQuery().getType());
+        List<Map<String, Object>> result = queryClient.query(url, queryWithResult.getQuery());
+        if (!QueryResultVerifier.compareResults(result, queryWithResult.getExpectedResults())) {
+          LOG.error(
+              "Failed while executing %s actualResults : %s",
+              queryWithResult,
+              jsonMapper.writeValueAsString(result)
+          );
+          failed = true;
+        } else {
+          LOG.info("Results Verified for Query %s", queryWithResult.getQuery().getType());
+        }
+      }
+
+      if (failed) {
+        throw new ISE("one or more queries failed");
+      }
+    }
+  }
+
+  private void testFailureQueries(List<QueryWithResults> queries, int timesToRun) throws Exception
+  {
+    for (QueryWithResults queryWithResult : queries) {
+      LOG.info("Running Query %s", queryWithResult.getQuery().getType());
+      try {
+        List<Map<String, Object>> result = queryClient.query(queryWithResult.getQuery());
+        throw new ISE(
+            "No error while querying[%s]",
+            queryWithResult.getQuery()
+        );
+      }
+      catch (ISE e) {
+        LOG.info("query failed, as expected");
```

Comment:
wouldn't you come here even if query was successful and ISE is really from the line 134?